design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,187 @@
+# Stacks Security and Isolation
+
+* Owner: Jared Watts (@jbw976)
+* Reviewers: Crossplane Maintainers
+* Status: Draft, revision 1.0
+
+## Background
+
+The current model for extending Crossplane with new functionality is scoped to a single type that is used in all scenarios and by all personas.
+All new types of functionality, from an end user wanting to install a simple application to an administrator wanting to install support for low level infrastructure types, are all treated the same.
+They have the same approach to security and access control and the artifacts that they install have the same scope of impact to the control plane.
+
+This one size fits all approach is not ideal though, the multiple scenarios and personas need finer grain control around security, access, and isolation.
+This design doc proposes a new approach for the Crossplane Stack Manager (SM) that will support this increased control for the installation of stacks.
+
+## Personas and Scenarios
+
+There are many different scenarios and types of users that may want to install new functionality into their control plane.
+This design proposes support for two high level personas/scenarios as an effective starting point.
+
+* **Platform:** An administrator of the control plane that is installing wide ranging support for lower level infrastructure and platform components,
+such as installing support for an entire cloud provider and all of its infrastructure and platform services.
+  * An example of this would be the `GCP` stack that contains a `Network` primitive and the `CloudSQL` resource.
+This persona would also be expected to define resource classes and other environment specific details.
+* **Application:** An application operator or end user of Crossplane.
+They are focused at a higher level and want to install the functionality of a single application, such as `GitLab` or `WordPress`.
+This persona is not expected to know or control any details about the environments managed by the Crossplane cluster, they simply need to consume their new application of choice.
+
```

Comment:
I think the usage of the terms `Platform` and `Application` might vary quite a bit between different teams and organizations. I wonder if we should use different terms here that more clearly identify the difference in isolation levels, vs. what the stack does or who uses it.

Reply:
It seems odd to me that, according to this document, a "platform stack" installs support for "low level infrastructure types". I could also conceive of certain "applications" such as GitLab or Jenkins being part of the "platform" that a platform-persona-type is providing to their customers.

Given the subjective nature of these concepts, I like the idea of differentiating by isolation or privilege level.